File: 2023-2024/sourcecode/mod4/lab12/RetrofitBasicsJava/app/src/main/java/com/example/retrofitbasicsjava/ui/PostViewHolder.java
package com.example.retrofitbasicsjava.ui;

import androidx.recyclerview.widget.RecyclerView;

import com.example.retrofitbasicsjava.data.Post;
import com.example.retrofitbasicsjava.databinding.RvItemBinding;

public class PostViewHolder extends RecyclerView.ViewHolder {
    private final RvItemBinding binding;

    public PostViewHolder(RvItemBinding binding) {
        super(binding.getRoot());
        this.binding = binding;
    }

    public void bind(Post item) {
        binding.title.setText(item.getTitle());
        binding.content.setText(item.getContent());
        binding.userId.setText(String.valueOf(item.getUserId()));
    }
}

File: 2023-2024/sourcecode/mod4/lab12/RetrofitBasicsKotlin/app/src/main/java/com/example/retrofitbasicskotlin/ui/PostViewHolder.kt
package com.example.retrofitbasicskotlin.ui

import androidx.recyclerview.widget.RecyclerView
import com.example.retrofitbasicskotlin.data.Post
import com.example.retrofitbasicskotlin.databinding.RvItemBinding

class PostViewHolder(private val binding: RvItemBinding) : RecyclerView.ViewHolder(binding.root) {
    fun bind(item: Post) {
        binding.apply {
            title.text = item.title
            content.text = item.content
            userId.text = item.userId.toString()
        }
    }
}